server/service/osquery.go

```diff
@@ -1165,7 +1165,7 @@ func (svc *Service) ingestDistributedQuery(ctx context.Context, host fleet.Host,
 		},
 		Rows: rows,
 	}
-	if failed {
```

Comment:
If the 'failed' param this is not required consider removing it.

Reply:
Changing from `failed` to `errMsg != ""` ensures that when the query is ran and data is successfully returned, we still can retrieve any error messages attached to it

Reply:
Do you still need the `failed` passed as a param to this function?

Reply:
Ooo, it looks like not. I updated the PR and removed the `failed` param from the function and all its tests, I QAed it, but if @lucasmrod or some other Go guru can confirm it looks good, that would be great

Reply:
Looks like it's only used locally here so safe.

`sharonkatz@Sharons-MacBook-Pro fleet % git grep ingestDistributedQuery
server/service/osquery.go:              err = svc.ingestDistributedQuery(ctx, *host, query, rows, failed, messages[query])
server/service/osquery.go:// ingestDistributedQuery takes the results of a distributed query and modifies the
server/service/osquery.go:func (svc *Service) ingestDistributedQuery(ctx context.Context, host fleet.Host, name string, rows []map[string]string, failed bool, errMsg string) error {
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "bad_name", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")
server/service/osquery_test.go: err := svc.ingestDistributedQuery(context.Background(), host, "fleet_distributed_query_42", []map[string]string{}, false, "")`